File: backend/app/models/organization.py
from sqlalchemy import Boolean, Column, Integer, String, Text, DateTime, ForeignKey
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship

from app.db.base import Base, BaseModel

class Organization(Base, BaseModel):
    """Модель организации"""
    
    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(255), nullable=False)
    code = Column(String(50), unique=True, nullable=False)
    description = Column(Text, nullable=True)
    org_type = Column(String(50), nullable=False)  # 'HOLDING', 'LEGAL_ENTITY', etc.
    parent_id = Column(Integer, ForeignKey("organization.id"), nullable=True)
    is_active = Column(Boolean, default=True, nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False)
    
    # Отношения
    parent = relationship("Organization", remote_side=[id], backref="children")
    divisions = relationship("Division", back_populates="organization")
    
    __tablename__ = "organization"  # Явно указываем имя таблицы 

File: frontend/src/types/organization.ts
// Типы данных для Организаций

// Базовый тип организации
export interface Organization {
  id: number;
  name: string;
  code: string;
  description: string | null;
  org_type: 'HOLDING' | 'LEGAL_ENTITY' | 'LOCATION';
  parent_id: number | null;
  is_active: boolean;
  created_at?: string;
  updated_at?: string;
  children?: Organization[];
}

// Тип для создания организации
export interface OrganizationCreateDto {
  name: string;
  code: string;
  description?: string | null;
  org_type: 'HOLDING' | 'LEGAL_ENTITY' | 'LOCATION';
  parent_id?: number | null;
  is_active?: boolean;
}

// Тип для обновления организации
export interface OrganizationUpdateDto {
  name?: string;
  code?: string;
  description?: string | null;
  org_type?: 'HOLDING' | 'LEGAL_ENTITY' | 'LOCATION';
  parent_id?: number | null;
  is_active?: boolean;
} 